### DISCUSSION TITLE
Use sessionStorage instead of localStorage

### DISCUSSION THREAD

**Discussion by idrm** [soofstad/react-oauth2-pkce#16] [Ideas]
Storing auth details in `sessionStorage` in place of `localStorage` will improve security since it wipes out the data when the authenticated user closes the browser tab/window. 

Alternately, a new config setting can be added to indicate the type of browser storage used, which would cover cases where it's OK to retain the details past the active session.

**Comment by soofstad** [comment]
Hi, glad you interested and involved in the library and it's improvement.
I'd welcome a feature where the users may explicitly set that secrets should be stored in "sessionStorage" instead of "localStorage".
But IMO it should not be the default, as the extra security benefits are minimal (still vulnerable to XSS, and access to an unlocked desktop is usually not protected against).
So, not much safer, but will increase the web app's loading time, and auth provider requests (this might  matter in scenarios where it's a pay-per-login billing model) .

What I would really like to have, is XSS safe token storage. That can be done via cookies, but I don't think that is supported by most authentication providers. If it's supported by _some_, a config parameter to enable this would be awsome.

An alternative to cookies could be what Auth0 mentions here; https://auth0.com/docs/secure/security-guidance/data-security/token-storage#browser-in-memory-scenarios Not sure how this would work, or if it's actually XSS safe.

If anyone is more knowledgeable on these topics I be glad to hear about any proposals.

**Comment by soofstad** [comment]
@idrm, and anyone else interested in this topic. The package supports using "sessionStorage" in place of "localStorage" with pull request #71, which will be available from version >=1.13
